tests/frontend/test_circuitspecs_chip2.py

```diff
@@ -0,0 +1,373 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Unit tests for the CircuitSpec class"""
+import textwrap
+
+import pytest
+import numpy as np
+import networkx as nx
+
+import blackbird
+
+import strawberryfields as sf
+import strawberryfields.ops as ops
+
+from strawberryfields.parameters import par_evaluate
+from strawberryfields.program_utils import CircuitError, list_to_DAG
+from strawberryfields.io import to_program
+from strawberryfields.utils import random_interferometer
+from strawberryfields.circuitspecs.chip2 import Chip2Specs, CircuitSpecs
+
+
+pytestmark = pytest.mark.frontend
+
+np.random.seed(42)
+
+
+def program_equivalence(prog1, prog2, compare_params=True, atol=1e-6, rtol=0):
```

Comment:
did this never get put in the main library??

Reply:
Nope :( I want to add it to `program_utils`, or make it a program method.

Reply:
This is still an open issue, so resolving this comment for now.